distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso

```diff
@@ -429,19 +429,29 @@ type Date
        Add the specified amount of time to this instant to get another date.
 
        Arguments:
-       - amount: The time duration to add to this instant.
+       - amount: The time duration to add to this instant. Can only be a
+        `Period` or a `Date_Period`.
 
        > Example
          Add 6 months to a local date.
 
              import Standard.Base.Data.Time.Duration
 
              example_add = Date.new 2020 + (Period.new months=6)
+
+       > Example
+         Add a month to a local date.
+
+             import Standard.Base.Data.Time.Date_Period
+
+             example_add = Date.new 2020 + Date_Period.Month
     + : Period -> Date ! (Time_Error | Illegal_Argument)
-    + self period =
-        case period of
-            _ : Period ->
+    + self amount =
+        case amount of
+            period : Period ->
                 Time_Utils.date_adjust self Time_Utils.AdjustOp.PLUS period.internal_period
+            date_period : Date_Period ->
+                self + date_period.to_period
             _ : Duration ->
                 Error.throw (Time_Error.Error "Date does not support adding/subtracting Duration. Use Period instead.")
```

Comment:
could move to type checking `+ self amount:(Period|Date_Period)` then can lose the default branch.